src/main.rs

```diff
@@ -373,6 +375,55 @@ fn reply_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
     0
 }
 
+/// show subcommand implementation
+///
+fn show_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // NOTE: we use boxed iterators since we might do a lot of dynamic wrapping
+    //       depending on command line parameters
+
+    // first, get us an iterator over all the commits
+    // NOTE: "issue" is a required parameter
+    let issue = try_or_1!(Oid::from_str(matches.value_of("issue").unwrap()));
+    let mut commits : Vec<(TreeGraphElemLine, Commit)> =
+        try_or_1!(repo.get_issue_revwalk(issue)).abort_on_err()
+            .map(|oid| repo.find_commit(oid)).abort_on_err()
+            .into_tree_graph()
+            .collect();
+
+    // Decide on the order in which the messages will be printed.
+    if matches.is_present("tree") {
+        // We want the commits in chronological order
+        commits.reverse();
+        for commit in commits.iter_mut() {
+            commit.0.reverse_marks();
+        }
+    };
+
+    // Transform the simple graph element line into an iterator over lines to
+    // print via multiple steps.
+    let graph = commits
+        .into_iter()
+        // expand the graph element lines for each message
+        .map(|commit| {
+            let mut elems = commit.0;
+            // offset the commit from the graph elements by adding an empty one
+            // in between
+            elems.append(TreeGraphElem::Empty);
+            (elems.commit_iterator(), commit.1)
+        })
+        // expand the message to a series of lines
+        .flat_map(|commit| commit.0.zip(vec![commit.1.id().to_string(), String::new()]))
+        // combine each line of graph elements and message
+        .map(|line| format!("{} {}", line.0, line.1));
+
+    // spawn a pager and write the graph
+    let mut pager = try_or_1!(programs::pager(try_or_1!(repo.config())));
```

Comment:
Maybe we should open the pager object before the whole iterator chain, for early-fail reasons.